### ISSUE TITLE
zsh ignores interactive shell and outer environment

### ISSUE THREAD

**Issue by christophmegusta** [robbyrussell/oh-my-zsh#5087]
when we do something like this when developing, the zsh is ignoring the sourced outer environment.
```
  # starting from bash or zsh doesnt matter
  source ./env/activate # from virtualenv  which makes our_command available
  which our_command # just fine
  zsh # with ohmyzsh installed     , also zsh -i doesnt make a difference, we always get a fresh shell
  which our_command # not found
  exit # now we are back in invoking (outer) shell
  zsh -c 'which our_command'    # this WORKS!
  zsh -c 'print `which our_command`; zsh -i' # this prints correct, but new zsh doesnt keep it
```
if i uninstall ohmyzsh and use regular zsh it is working as expected.

it seams that zsh is always starting a fresh login zsh. but only when in interactive mode. if we use for example -c , its working.

also this did work a few month ago without any problems. Just after one of the ohmyzsh upgrades, it did break. maybe its a new feature and we need to set somewhere a parameter?

**Comment by mcornella** [comment]
Commands are not passed from session to session, unlike exported variables. The `zsh -c` examples shouldn't work, so there must be some file in zsh's startup sequence that contains the sourcing, which is somehow not triggered when running on an interactive session.

Have a look at [zsh's startup files](http://zsh.sourceforge.net/Guide/zshguide02.html#l9) to check where in those you have the culprit. If you can't figure it out, paste them here or on a gist and I'll try to diagnose it.

**Comment by christophmegusta** [comment]
Thank you for your fast answer.
I could isolate the problem to ~/.zshrc. Further it seams to be the PATH export also in combination with sourcing of oh-my-zsh.

My os is Mac OS 10.11.4
and zsh version: zsh 5.0.8 (x86_64-apple-darwin15.0)
and ~/.zshrc:
```
export ZSH=/Volumes/myuser/Home/.oh-my-zsh

ZSH_THEME="robbyrussell"

plugins=(git)

# THIS breaks it (what i had before)
export PATH="/Volumes/myuser/Home/bin:/Volumes/myuser/Home/Library/Android/sdk/platform-tools:/Volumes/myuser/Home/Library/Android/sdk/tools:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/MacGPG2/bin"
# THIS fixes it
#export PATH=$PATH:"/Volumes/myuser/Home/bin:/Volumes/myuser/Home/Library/Android/sdk/platform-tools:/Volumes/myuser/Home/Library/Android/sdk/tools:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/MacGPG2/bin"

# if i comment this line the showing of virtualenvs prompt works as expected
source $ZSH/oh-my-zsh.sh
```

Though i never edited this file until now (just stripped comments for this ticket). Before i did open this ticket i removed the ~/.zshrc  and uninstalled oh-my-zsh and then reinstalled it again. So zsh + ohmyzsh did create it fresh.

Also my "fix" , prepending the $PATH: does only fix the commands. What still is not working is the shell prompt styling from Python's virtualenv.
(Though it is working perfectly if i open a zsh and then source the activate script inside)

**Comment by mcornella** [comment]
Ok thanks for the follow up. The PATH definition has been a pain in the ass for quite some time, I'm working on refactoring the installer so that this and other stuff works correctly.

**Comment by mcornella** [comment]
Closing this because `$PATH` mangling is a known problem. It should be fixed for new installations on #5169.

**Comment by christophmegusta** [comment]
thanks for fixing this. gladly can confirm its working for us now.
